### DISCUSSION TITLE
Change the way to use components

### DISCUSSION THREAD

**Discussion by cavasinf** [kevinpapst/TablerBundle#146] [Ideas]
Take a HUGE look at this ➡️ https://ux.symfony.com/twig-component

AKA : https://symfony.com/bundles/ux-twig-component/current/index.html

```php
use Symfony\UX\TwigComponent\Attribute\ExposeInTemplate;

#[AsTwigComponent('alert')]
class AlertComponent
{
    #[ExposeInTemplate]
    private string $message; // available as `{{ message }}` in the template

    #[ExposeInTemplate('alert_type')]
    private string $type = 'success'; // available as `{{ alert_type }}` in the template

    #[ExposeInTemplate(name: 'ico', getter: 'fetchIcon')]
    private string $icon = 'ico-warning'; // available as `{{ ico }}` in the template using `fetchIcon()` as the getter

    /**
     * Required to access $this->message
     */
    public function getMessage(): string
    {
        return $this->message;
    }

    /**
     * Required to access $this->type
     */
    public function getType(): string
    {
        return $this->type;
    }

    /**
     * Required to access $this->icon
     */
    public function fetchIcon(): string
    {
        return $this->icon;
    }

    #[ExposeInTemplate]
    public function getActions(): array // available as `{{ actions }}` in the template
    {
        // ...
    }

    #[ExposeInTemplate('dismissable')]
    public function canBeDismissed(): bool // available as `{{ dismissable }}` in the template
    {
        // ...
    }

    // ...
}
```

**Comment by cavasinf** [comment]
@kevinpapst IDK if you are tagged when new conversation

**Reply by kevinpapst** [reply]
Yes. And I am not convinced... 
Pro: easier to test and a unique way of calling them
Con: HUUUUUUGE REFACTORING TIME

Probably for a version 2 ?

**Reply by cavasinf** [reply]
I'm not saying we should jump in head first, but I'm doing my "technology watch".

I don't like that with the current state we don't have automatic completion for components.
But I'm pretty sure you won't get either with this component "way".
AND maybe not even with `CTRL + Click` NOR `middle mouse click` to go to the reference (depends on the Symfony plugin in PhpStorm).

I'm going to try it on some of my recent projects.

**Reply by kevinpapst** [reply]
Ah okay, I was afraid that you suggest that we should use it 😁 changing this to components sounds like a massive effort.
 
I had a deeper look at the UX components multiple times. When they came out and I follow the progress regularly. But for some reasons, I am not yet sure if I really need that.
